experimental/dds/tree2/src/feature-libraries/contextuallyTyped.ts

```diff
@@ -234,6 +235,38 @@ export type ContextuallyTypedNodeData =
  */
 export type ContextuallyTypedFieldData = ContextuallyTypedNodeData | undefined;
 
+/**
+ * Information needed to interpret a subtree described by {@link ContextuallyTypedNodeData} and {@link ContextuallyTypedFieldData}.
+ * @alpha
+ */
+export interface TreeDataContext {
+	/**
+	 * Schema for the document which the tree will be used in.
+	 */
+	readonly schema: SchemaDataAndPolicy;
+
+	/**
+	 * Procedural data generator for fields.
+	 * Fields which provide generators here can be omitted in the input contextually typed data.
+	 *
+	 * @remarks
+	 * TODO:
+	 * For implementers of this which are not pure (like identifier generation),
+	 * order of invocation should be made consistent and documented.
+	 * This will be important for identifier elision optimizations in tree encoding for session based identifier generation.
+	 */
+	fieldSource?(key: FieldKey, schema: FieldStoredSchema): undefined | FieldGenerator;
+}
+
+type GetFieldGenerator = (key: FieldKey, schema: FieldStoredSchema) => undefined | FieldGenerator;
+export const defaultFieldSource: GetFieldGenerator = () => undefined;
```

Comment:
I think we might not need this anymore, since `fieldSource` can be undefined instead.

Reply:
removed `defaultFieldSource` and `GetFieldGenerator`